### ISSUE TITLE
Backquery

### ISSUE THREAD

**Issue by shiffman** [shiffman/Neural-Network-p5#5]
I'd like to demonstrate a "back query" (feeding a desired output backwards through the network and getting pixels as a result) with this example. In @makeyourownneuralnetwork, the back query concept is explained and a sample image is provided (for the digit 0):

<img width="263" alt="screen shot 2017-04-12 at 3 57 43 pm" src="https://cloud.githubusercontent.com/assets/191758/24976840/fed3da20-1f98-11e7-82ec-94e3e4574c2e.png">

Perhaps I'm too impatient but after running through 10,000 training images, my backquery for zero looks like:

<img width="169" alt="screen shot 2017-04-12 at 4 07 10 pm" src="https://cloud.githubusercontent.com/assets/191758/24977157/21c9e6fe-1f9a-11e7-8124-7547978d06df.png">

I think I'm missing something. Code in progress is in the `backquery` branch:

https://github.com/shiffman/Neural-Network-p5/blob/backquery/nn.js#L127

**Comment by makeyourownneuralnetwork** [comment]
I'll take a look at the code tomorrow (but I'm no javascript expert).

What seems immediately suspect is that the learning doesn't seem to have completed sufficiently .. over time the shape should become "smooth" in the different colours. The top image looks like it is a result of a Gaussian blur - the bottom one doesn't yet.

Having said that even 10,000 training examples should be sufficient to have formed a good shape so something is off. Does 10,000 training examples (with whatever other parameters e.g. number of nodes, learning rate, etc) give you an overall performance of around 95% for the moist 10,000 test set? If yes.. this is puzzling.

**Comment by makeyourownneuralnetwork** [comment]
Here is someone else's animated visualisation of the network learning ... it gets smooth pretty quick.

[https://www.youtube.com/watch?v=1vYaBCM2WOQ](https://www.youtube.com/watch?v=1vYaBCM2WOQ)
